app/uk/gov/hmrc/filetransmission/services/queue/MongoBackedWorkItemService.scala

```diff
@@ -77,7 +79,20 @@ class MongoBackedWorkItemService @Inject()(
   override def clearQueue(): Future[Boolean] = repository.clearRequestQueue()
 
   private def processWorkItem(
-      workItem: WorkItem[TransmissionRequestEnvelope]): Future[Unit] = {
+      workItem: WorkItem[TransmissionRequestEnvelope],
+      processedAt: Instant = clock.instant
+  ): Future[Unit] = {
+
+    // Update the work item with the latest FailedDeliveryAttempt, then update the status.
+    def updateStatusForFailedDeliveryAttempt(status: ResultStatus,
+                                             availableAt: Option[DateTime],
+                                             ex: Throwable): Future[Boolean] = {
+      val b1 = repository.updateWorkItemBody(workItem.id, workItem.item.withFailedDeliveryAttempt(FailedDeliveryAttempt(processedAt, ex.getMessage)))
```

Comment:
Shouldn't we run these operations in sequence? I don't know what's happening in markAs but isn't there a risk of race condition?